dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityOperationParamsMapper.java

```diff
@@ -166,6 +169,16 @@ private List<TrackedEntityType> filterAndValidateTrackedEntityTypes(User user)
     return trackedEntityTypes;
   }
 
+  private List<Program> getPrograms(Program program, User user) {
+    if (program == null) {
+      return programService.getAllPrograms().stream()
```

Comment:
Here we could already filter out event programs or even better we could create a method in `ProgramService` to get all tracker programs.

Reply:
I agree it would make more sense to have a method in `ProgramService` that provides us with tracker programs directly.
Thing is, as you know, that service has some methods that validate metadata access and some that don't.
As we discussed with Viet, it might make more sense to create our own tracker program service and make sure we just add methods in there that are metadata validated. I'd rather do that than continue adding methods in the current program service. That could be a fairly big refactor and it might need a team discussion first , so I'll leave it for another ticket.